dashboard/app/controllers/puzzle_ratings_controller.rb

```diff
@@ -0,0 +1,15 @@
+class PuzzleRatingsController < ApplicationController
+  def create
+    status_code = :unauthorized
+    if PuzzleRating.enabled?
+      valid_request = PuzzleRating.create(
```

Comment:
you may want to handle the mysql uniqueness constraint violation exception here. The model level validation will catch most cases but it there are possible race conditions where it ends up in the database (who does not fall victim to race conditions). Some examples:

https://github.com/code-dot-org/code-dot-org/search?utf8=%E2%9C%93&q=Mysql2%3A%3AError

I think doing the same retry pattern as we use elsewhere will give you an not-valid? record which makes sense.